src/nb_type.cpp

```diff
@@ -222,6 +227,62 @@ int nb_type_init(PyObject *self, PyObject *args, PyObject *kwds) {
     return 0;
 }
 
+/// dynamic_attr: Support for `d = instance.__dict__`.
+extern "C" inline PyObject *nb_get_dict(PyObject *self, void *) {
```

Comment:
You can drop the extern "C" annotation and make the function `static`. Ditto for all the others below.